modules/standard/IO.chpl

```diff
@@ -4345,14 +4354,25 @@ proc fileReader.advance(amount:int(64)) throws {
 }
 
 /*
-   Move a fileWriter offset forward.
+   Move a :record:`fileWriter` offset forward.
+
+   This routine will populate the ``fileWriter``'s buffer as the offset is moved
+   forward by ``amount`` bytes. If the file contains data after its current
+   offset, those data will be read into the buffer. Otherwise, the buffer will
```

Comment:
I'm not so sure this is right. If you mark/write/revert/seek then I would expect you could see data that you wrote earlier. And it's possible when using memory mapped files that you might end up with data from the file. Which leads me to thinking that this should be described as "the data could be any of these things". But, if we did that, I'm not so sure that this method would be useful. Do you happen to know where this method is used, if it is used at all?

I think I'd still lean towards documenting the uncertainty for now, even though it might make this method useless. I'd consider switching it to always fill in 0 bytes in the future; but that would be a non-breaking change with the documented uncertainty.

Reply:
Okay, I was not aware of that. I'm not seeing many tests that use it, and it doesn't look like it's used in Arkouda. I'll go with the uncertainty in the docs for now.